torch/csrc/jit/interned_strings.h

```diff
@@ -3,146 +3,252 @@
 #include <stdint.h>
 #include <string>
 #include <unordered_map>
+#include <algorithm>
 
 #include "torch/csrc/jit/generated/aten_interned_strings.h"
 
 namespace torch { namespace jit {
 
-// JIT symbols are synthetic operators that occur only in the JIT IR
+// Every symbol is classified in a namespace, specifying what kind of symbol it
+// is.  Unsigned char to ensure widening to unique_t (also an unsigned type)
+enum class SymbolNamespace : unsigned char {
+  onnx  = 'o',
+  prim  = 'p',
+  aten  = 't',
+  // NB: ONNX and ATen attributes all live in a global namespace, as
+  // their interpretation depends on the operator name (which is namespaced)
+  attr  = 'a',
+  // TODO: eliminate me
+  scope = 's'
+};
+
+namespace detail {
+
+  // TODO: programatically generate this
+  inline std::string valid_namespaces_str() {
+    return "'onnx', 'prim', 'aten', 'attr', 'scope'";
+  }
+
+}
+
+// Primitive symbols are synthetic operators that occur only in the IR
 // and don't have corresponding implementations in ATen.
 //
 // TODO: We need documentation for all of these symbols.
+//
+// TODO: Consider moving the synthetic onnx operators to their own
+// namespace.
 
-#define FORALL_JIT_SYMBOLS(_) \
-_(Assign) \
-_(Constant) \
-_(CppOp) \
-_(Drop) \
-_(Eval) \
-_(Expand) \
-_(FusionGroup) \
-_(GraphExecutor) \
-_(If) \
-_(Jump) \
-_(JumpNZ) \
-_(JumpZ) \
-_(Load) \
-_(Loop) \
-_(Param) \
-_(Placeholder) \
-_(Print) \
-_(PythonOp) \
-_(ReplaceIfUndef) \
-_(Reverse) \
-_(Return) \
-_(Store) \
-_(Undefined) \
-_(__JIT_END)
+#define FORALL_PRIM_SYMBOLS(_) \
+_(prim, Assign) \
+_(prim, Constant) \
+_(prim, CppOp) \
+_(prim, Drop) \
+_(prim, Eval) \
+_(prim, Expand) /* onnx */ \
+_(prim, FusionGroup) \
+_(prim, GraphExecutor) \
+_(prim, If) \
+_(prim, Jump) /* debug */ \
+_(prim, JumpNZ) /* debug */ \
+_(prim, JumpZ) /* debug */ \
+_(prim, Load) \
+_(prim, Loop) \
+_(prim, Param) \
+_(prim, PackPadded) /* onnx */ \
+_(prim, PadPacked) /* onnx */ \
+_(prim, Placeholder) /* debug */ \
+_(prim, Print) \
+_(prim, PythonOp) \
+_(prim, ReplaceIfUndef) \
+_(prim, Reverse) \
+_(prim, Return) \
+_(prim, Store) \
+_(prim, Undefined) \
+/* end */
 
 // Workaround for some not-yet-defined ATen symbols, see
 //  - __not__: https://github.com/pytorch/pytorch/issues/5495
 //  - ones, zeros: https://github.com/pytorch/pytorch/issues/5496
 
 #define FORALL_ATEN_EXTRA_SYMBOLS(_) \
-_(__not__) \
-_(ones) \
-_(zeros) \
-_(__ATEN_EXTRA_END)
+_(aten, __not__) \
```

Comment:
lol, but why

Reply:
"Because the frontend is using this symbol" :P

Reply:
haha right

Reply:
Do we desugar it somehow? We don't have this op in the dispatch I think

Reply:
Don't ask me! I didn't add it.